packages/flutter/lib/src/material/button_style.dart

```diff
@@ -315,6 +323,22 @@ class ButtonStyle with Diagnosticable {
   /// ```
   final InteractiveInkFeatureFactory? splashFactory;
 
+  /// Creates a widget that becomes the child of the button's [Material]
+  /// and whose child is the rest of the button, including the button's
+  /// `child` parameter.
+  ///
+  /// By default the returned widget is clipped to the Material's [ButtonStyle.shape].
```

Comment:
"By default" makes it sound like that this can be changed? Maybe leave a breadcrumb to tell people how this can be changed?

Reply:
Good point, yes.